rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -536,6 +554,112 @@ void saveOldFederation_after_RSKIP123_should_save_null() {
         );
     }
 
+    private static Stream<Arguments> provideFederationBtcUTXOsTestArguments() {
+        return Stream.of(
+            Arguments.of(false, false, NetworkParameters.ID_TESTNET),
+            Arguments.of(false, false, NetworkParameters.ID_MAINNET),
+            Arguments.of(false, true, NetworkParameters.ID_TESTNET),
+            Arguments.of(false, true, NetworkParameters.ID_MAINNET),
+            Arguments.of(true, false, NetworkParameters.ID_TESTNET),
+            Arguments.of(true, false, NetworkParameters.ID_MAINNET),
+            Arguments.of(true, true, NetworkParameters.ID_TESTNET),
+            Arguments.of(true, true, NetworkParameters.ID_MAINNET)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideFederationBtcUTXOsTestArguments")
+    void testGetNewFederationBtcUTXOsWithCombinationsOfRSKIPsAndNetworks(boolean isRskip284Active, boolean isRskip293Active, String networkId) {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(isRskip284Active);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(isRskip293Active);
+
+        NetworkParameters networkParameters = NetworkParameters.fromID(networkId);
+
+        Address btcAddress = new Address(testnetBtcParams, Hex.decode("4a22c3c4cbb31e4d03b15550636762bda0baf85a"));
```

Comment:
Do you need this particular address. There's a method in BitcoinTestUtils called `createP2PKHAddress`.

Also, why use testnet params? This test is executed both for testnet and mainnet environments

Reply:
Updated.